topics/dev/tutorials/core-contributing/tutorial.md

```diff
@@ -121,50 +171,112 @@ The relational database tables consumed by Galaxy are defined in ``lib/galaxy/mo
+7. Use the `relationship` function to define an association between your new model and the `User` model.
+
+To define a regular `Column` attribute you must include the datatype:
+
+```python
+foo = Column(Integer)
+```
+
+To define an attribute that is the primary key, you need to include the `primary_key` argument:
+(a primary key is one or more columns that uniquely identify a row in a database table)
+
+```python
+id = Column(Integer, primary_key=True)
+```
+
+To define an attribute that is a foreign key, you need to reference the associated table + its primary key column using the `ForeignKey` construct (the datatype will be derived from that column, so you don't have to include it):
+
+```python
+bar_id = Column(ForeignKey("bar.id"))
+```
+
+To define a relationship between tables, you need to set it on *both* tables:
 
-{% include topics/dev/tutorials/core-contributing/mapping.py_diff.md %}
+```python
+class Order(Base):
+    …
+    items = relationship("Item", back_populates="order")
 
-The Python model objects used by Galaxy corresponding to these tables are
-defined in ``lib/galaxy/model/__init__.py``.
+class Item(Base):
+    …
+    order_id = Column(ForeignKey("order.id")
+    awesome_order = relationship("Order", back_populates="items")
+    # relationship not named "order" to avoid confusion: this is NOT the table name for Order
+```
 
-Modify ``lib/galaxy/model/__init__.py`` to add a model class
-called ``UserFavoriteExtension`` as described above.
+Now modify `lib/galaxy/model/__init__.py` to add a model class called `UserFavoriteExtension` as described above.
 
 {% include topics/dev/tutorials/core-contributing/__init__.py_diff.md %}
 
-## Migrations
+# Migrations
 
-There is one last database issue to consider before moving on to considering the API.
-Each successive release of Galaxy requires recipes for how to migrate old database schemes
-to updated ones. These recipes are called versions and are currently implemented using SQLAlchemy
- Migrate. These versions are stored in ``lib/galaxy/model/migrate``.
+There is one last database issue to consider before moving on to considering the API. Each successive release of Galaxy requires recipes for how to migrate old database schemas to updated ones. These recipes are called versions, or revisions, and are implemented using Alembic. 
 
-Each of these versions is prefixed with a 4-digit number (e.g. ``0125``) to specify the linear
-order these migrations should be applied in.
+Galaxy's data model is split into the galaxy model and the install model. These models are persisted in one combined database or two separate databases and are represented by two migration branches: "gxy" (the galaxy branch) and "tsi" (the tool shed install branch). Schema changes for these branches are defined in these revision modules:
+- `lib/galaxy/model/migrations/alembic/versions_gxy` (galaxy model)
+- `lib/galaxy/model/migrations/alembic/versions_tsi` (install model)
```

Comment:
I appreciate the work you did to maintain the old functionality here but you keep calling out these separate models and no one will ever update them. They are essentially not used and not useful. The next time they will be updated is when we rip them out. I think it would be better to just to refer to ``lib/galaxy/model/migrations/alembic/versions_gxy`` as the Galaxy migrations.

Reply:
I'll update this here and will address this in the documentation (migrations readme, documentation in scripts etc.)